src/aro/Preprocessor.zig

```diff
@@ -1462,6 +1466,15 @@ fn handleBuiltinMacro(pp: *Preprocessor, builtin: RawToken.Id, param_toks: []con
     }
 }
 
+fn getPasteArgs(args: []const []const Token, end: u32) []const Token {
+    const placemarker = &[1]Token{tokFromRaw(.{ .id = .placemarker, .source = .generated })};
+    if (args[end].len == 0) return placemarker;
```

Comment:
How does this actually work? It looks like it's returning an address of a stack variable - is it because it's a comptime-known constant and it's getting hoisted up to some static-lifetime?

Reply:
`tokFromRaw` is being called at runtime so it is indeed returning an address of a stack variable.

Reply:
I'm surprised it's not crashing then, lol. In any of the modes I tried, at least. I rewrote it so `getPasteArgs` returns an optional instead.